File: SPCServer/springboot/src/main/java/com/lcaohoanq/shoppe/domain/otp/IOtpService.java
package com.lcaohoanq.shoppe.domain.otp;

import java.util.Optional;

public interface IOtpService {

    Otp createOtp(Otp otp);
    void disableOtp(long id);
    Optional<Otp> getOtpByEmailOtp(String email, String otp);
    void setOtpExpired();
    String generateOtp();
}

File: auth-service/src/main/kotlin/com/lcaohoanq/authservice/domains/otp/IOtpService.kt
package com.lcaohoanq.authservice.domains.otp

import com.lcaohoanq.authservice.domains.user.User
import com.lcaohoanq.common.dto.OtpPort

interface IOtpService {
    fun createOtpFor(user: User, otpRequest: OtpPort.OtpReq)
    fun disableOtp(id: Long)
    fun getOtpByEmailOtp(email: String, otp: String): Otp?
    fun setOtpExpired()
    fun generateOtp(): String
    fun getAllOtps(): List<OtpPort.OtpRes>

}
